libwallet/internal/politeia/politeia.go

```diff
@@ -90,28 +91,45 @@ func (p *Politeia) saveOrOverwiteProposal(proposal *Proposal) error {
 }
 
 // GetProposalsRaw fetches and returns a proposals from the db
-func (p *Politeia) GetProposalsRaw(category int32, offset, limit int32, newestFirst bool) ([]Proposal, error) {
-	return p.getProposalsRaw(category, offset, limit, newestFirst, false)
+func (p *Politeia) GetProposalsRaw(category int32, offset, limit int32, newestFirst bool, key string) ([]Proposal, error) {
+	return p.getProposalsRaw(category, offset, limit, newestFirst, false, key)
 }
 
-func (p *Politeia) getProposalsRaw(category int32, offset, limit int32, newestFirst bool, skipAbandoned bool) ([]Proposal, error) {
+func (p *Politeia) getProposalsRaw(category int32, offset, limit int32, newestFirst bool, skipAbandoned bool, searchKey string) ([]Proposal, error) {
 	var query storm.Query
+	searchKey = strings.TrimSpace(strings.ToLower(searchKey))
 	switch category {
 	case ProposalCategoryAll:
```

Comment:
See if this code achieves the same results for preparing the query. It's shorter and therefore easier to understand:
```
	matcher := q.Eq("Category", category)
	if category == ProposalCategoryAll {
		if skipAbandoned {
			matcher = q.Not(q.Eq("Category", ProposalCategoryAbandoned))
		} else {
			matcher = q.True()
		}
	}

	query := p.db.Select(matcher)
	if searchKey != "" {
		query = p.db.Select(matcher, q.Re("LowerName", "^"+searchKey))
	}
```